src/main/java/org/osiam/storage/query/ExtensionQueryField.java

```diff
@@ -48,7 +49,11 @@ public Predicate addFilter(Root<UserEntity> root, FilterConstraint constraint,
         if (constraint != FilterConstraint.PRESENT && (field.getType() == ExtensionFieldType.INTEGER ||
                 field.getType() == ExtensionFieldType.DECIMAL)) {
 
-            value = numberPadder.pad(value);
+            try {
+                value = numberPadder.pad(value);
+            } catch (IllegalArgumentException ex) {
+                throw new InvalidFilterException(ex.getMessage());
```

Comment:
It might help to track down bugs when you see the usual giant stacktrace in the log. Does this error case even get logged?